app/controllers/concerns/generic_crud.rb

```diff
@@ -0,0 +1,61 @@
+module GenericCrud
+  extend ActiveSupport::Concern
+
+  def index
+    @entities = klass.includes(:answers, :comments).page(page).per(limit)
+
+    render json: @entities
+  end
+
+  def create
+    @entity = klass.new(**params_attributes, **options)
+    @entity.save!
+
+    render json: @entity
+  end
+
+  def show
+    render json: send(resource)
+  end
+
+  # TODO: enforce user authorization
+  def update
+    redirect_to root_path unless valid_user?
+    send(resource).update!(params_attributes)
```

Comment:
why use send here?

Reply:
I need to call `question` getter

https://github.com/lokendrasingh-1mg/StackOverflowRailsClone/blob/58fe7a50226f9b5fa68dc36f9242a8a7909ff271/app/controllers/questions_controller.rb#L47-L49

Its also required for `answer`

https://github.com/lokendrasingh-1mg/StackOverflowRailsClone/blob/58fe7a50226f9b5fa68dc36f9242a8a7909ff271/app/controllers/answers_controller.rb#L38-L40

Since its using `params[:id]`, I can provide a generic method in the concern itself